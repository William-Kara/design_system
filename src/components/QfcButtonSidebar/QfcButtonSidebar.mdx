import { ArgTypes, Meta } from '@storybook/blocks';
import QfcButtonSidebar from '@/components/QfcButtonSidebar/QfcButtonSidebar.vue';

<Meta title="Elements/ButtonSidebar" />

## Description

The QfcButtonSidebar component is a button that can be used in a sidebar menu to navigate to different pages in your application.

## Props

The following props are available for the QfcButtonSidebar component:

<ArgTypes of={QfcButtonSidebar} />
