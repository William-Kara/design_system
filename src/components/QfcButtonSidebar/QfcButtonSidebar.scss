.qfc-btn-sidebar {
  width: 100%;
  height: 4.31rem;
  display: flex;
  align-items: center;
  padding: $q-generic-space-4;
  background-color: $background;
  color: $grey-manatee;
  border-radius: $br-xs;
  border: $primary;
  text-align: left;

  &__content {
    display: flex;
    align-items: center;
    gap: $q-generic-space-3;

    &__icon {
      cursor: pointer;
    }
    &__label {
      font-size: $fs-xs;
    }
  }

  &:hover {
    background-color: $black-lighten;
    color: $white;
  }

  &.router-link-exact-active {
    color: $primary;
    background-color: $black-lighten;
  }
}

.qfc-btn-sidebar,
.qfc-btn-sidebar * {
  cursor: pointer;
}
