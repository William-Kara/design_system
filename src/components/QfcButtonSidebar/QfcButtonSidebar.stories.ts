import ButtonSidebar from './QfcButtonSidebar.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof ButtonSidebar> = {
  component: ButtonSidebar,
  title: 'Elements/ButtonSidebar',
  argTypes: {
    icon: { control: 'text', required: true },
    label: { control: 'text', required: true },
    to: { control: 'text', required: true },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    icon: 'space_dashboard',
    label: 'Dashboard',
    to: '/',
  },
};
