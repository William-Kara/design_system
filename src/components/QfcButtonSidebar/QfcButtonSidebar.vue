<template>
  <router-link :to="to" class="qfc-btn-sidebar" role="link">
    <span class="qfc-btn-sidebar__content">
      <span class="qfc-btn-sidebar__content__icon material-icons-outlined">{{ icon }}</span>
      <label class="qfc-btn-sidebar__content__label">{{ label }}</label>
    </span>
  </router-link>
</template>

<script lang="ts" setup>
interface Props {
  icon: string;
  label: string;
  to: string;
}

defineProps<Props>();
</script>

<style lang="scss" scoped src="./QfcButtonSidebar.scss" />
