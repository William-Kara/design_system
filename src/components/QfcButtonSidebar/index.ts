import QfcButtonSidebar from './QfcButtonSidebar.vue';

export { QfcButtonSidebar };
export default QfcButtonSidebar;
