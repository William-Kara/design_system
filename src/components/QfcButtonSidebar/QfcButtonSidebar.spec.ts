import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcButtonSidebar from './QfcButtonSidebar.vue';

let wrapper: any = null;

const propsValues = { icon: 'space_dashboard', label: 'Dashboard', to: '/dashboard' };

beforeEach(() => {
  wrapper = shallowMount(QfcButtonSidebar, {
    propsData: propsValues,
  });
});

describe('QfcButtonSidebar Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
