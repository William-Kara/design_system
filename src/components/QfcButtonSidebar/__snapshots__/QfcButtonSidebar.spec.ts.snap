// Vitest Snapshot v1

exports[`QfcButtonSidebar Component > should render correctly 1`] = `"<router-link data-v-cb1c8373=\\"\\" to=\\"/dashboard\\" class=\\"qfc-btn-sidebar\\" role=\\"link\\"><span data-v-cb1c8373=\\"\\" class=\\"qfc-btn-sidebar__content\\"><span data-v-cb1c8373=\\"\\" class=\\"qfc-btn-sidebar__content__icon material-icons-outlined\\">space_dashboard</span><label data-v-cb1c8373=\\"\\" class=\\"qfc-btn-sidebar__content__label\\">Dashboard</label></span></router-link>"`;
